layout/layout_test.go

```diff
@@ -0,0 +1,486 @@
+package layout_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/buildpacks/imgutil"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+
+	"github.com/buildpacks/imgutil/layout"
+
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	h "github.com/buildpacks/imgutil/testhelpers"
+)
+
+func TestImage(t *testing.T) {
+	spec.Run(t, "LayoutImage", testImage, spec.Report(report.Terminal{}))
+}
+
+func testImage(t *testing.T, when spec.G, it spec.S) {
+	var (
+		testImage           v1.Image
+		tmpDir              string
+		testDataDir         string
+		imagePath           string
+		fullBaseImagePath   string
+		sparseBaseImagePath string
+		err                 error
+	)
+
+	it.Before(func() {
+		// creates a v1.Image from a remote repository
+		testImage = h.CreateRemoteImage(t)
+
+		// creates the directory to save all the OCI images on disk
+		tmpDir, err = os.MkdirTemp("", "layout")
+		h.AssertNil(t, err)
+
+		// global directory and paths
+		testDataDir = filepath.Join("testdata", "layout")
+		fullBaseImagePath = filepath.Join(testDataDir, "busybox")
+		sparseBaseImagePath = filepath.Join(testDataDir, "busybox-sparse")
+	})
+
+	it.After(func() {
+		// removes all images created
+		os.RemoveAll(tmpDir)
+	})
+
+	when("#Save", func() {
+		it.After(func() {
+			os.RemoveAll(imagePath)
+		})
+
+		when("#FromBaseImage", func() {
+			it.Before(func() {
+				imagePath = filepath.Join(tmpDir, "save-from-base-image")
+			})
+
+			when("additional names are provided", func() {
+				it("creates an image with all the layers from the underlying image and org.opencontainers.image.ref.name annotation", func() {
```

Comment:
As mentioned elsewhere, maybe we can reduce complexity by not caring about additional names, unless there is a reason to care.